artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/wireformat/ReplicationSyncFileMessage.java

```diff
@@ -197,6 +232,15 @@ public void decodeRest(final ActiveMQBuffer buffer) {
       }
    }
 
+   private void readFile(ByteBuffer buffer) {
```

Comment:
You can create directly the FileChannel without using FileInputStream

Reply:
Could you elaborate this? I thought FileChannel can only be created by FileInputStream/FileOutputStream/RandomAccessFile.

Reply:
FileChannel::open